static/src/javascripts/projects/commercial/modules/brazeBanner.js

```diff
@@ -43,15 +43,17 @@ type InAppMessage = {
     },
 };
 
+type InAppMessageButton = any;
```

Comment:
Maybe implementing it according to [from Appboy / Braze’s documentation][appboy.doc] would be helpful.

[appboy.doc]: https://js.appboycdn.com/web-sdk/latest/doc/classes/appboy.inappmessagebutton.html

Reply:
Thanks for looking over this! It's an odd one; essentially, for our purposes, we will never need to use the first 6 parameters - so it felt unnecessary to provide type information for them. I'll have a think about this.

Reply:
It does feel like an inelegant solution at the moment.

Reply:
Maybe one option would be to wrap it in a function that takes in an object instead?

Reply:
If we did want to create add a Flow type definition for this then I think it would look something like:

```js
class InAppMessageButton {
  constructor(
    text: string,
    backgroundColor?: number,
    textColor?: number,
    borderColor?: number,
    clickAction?: ClickAction,
    uri?: string,
    id?: number): InAppMessageButton;
}
```

Which would also leave us with `ClickAction` to define, which looks like this apparently: https://js.appboycdn.com/web-sdk/latest/doc/classes/appboy.inappmessage.html#clickaction.

Reply:
I had a go at that Tom but didn't have any luck - after a few attempts I came up with something a bit different which Flow is happy with.

```javascript
type InAppMessageButtonObject = (
    text?: string | null,
    backgroundColor?: number| null,
    textColor?: number | null,
    borderColor?: number | null,
    clickAction?: ClickAction | null,
    uri?: string | null,
    id: number) => void;
```
[...]
```javascript
type AppBoy = {
    initialize: (string, any) => void,
    subscribeToInAppMessage: (InAppMessageCallback) => {},
    changeUser: (string) => void,
    openSession: () => void,
    logInAppMessageButtonClick: (InAppMessageButtonObject, InAppMessage) => void,
    logInAppMessageImpression: (InAppMessage) => void,
    InAppMessageButton: (null, null, null, null, null, null, number) => InAppMessageButtonObject,
};
```

Reply:
Now updated to be a bit more precise in terms typing, and consistent with the types used in DCR from Braze's type library:

```javascript
type InAppMessageButtonInstance = {
    text: string,
    backgroundColor?: number,
    textColor?: number,
    borderColor?: number,
    clickAction?: ClickAction,
    uri?: string,
    id?: number
}

type InAppMessageCallback = (InAppMessage) => void;

type AppBoy = {
    initialize: (string, any) => void,
    subscribeToInAppMessage: (InAppMessageCallback) => {},
    changeUser: (string) => void,
    openSession: () => void,
    logInAppMessageButtonClick: (InAppMessageButtonInstance, InAppMessage) => void,
    logInAppMessageImpression: (InAppMessage) => void,
    InAppMessageButton: (string, ?number, ?number, ?number, ?ClickAction, ?string, ?number) => InAppMessageButtonInstance,
};
```